adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/SniSSLSocketFactory.java

```diff
@@ -115,18 +120,34 @@ public Socket createLayeredSocket(Socket socket, String target, int port, HttpCo
     }
 
     private Socket applySNI(final Socket socket, String hostname) {
+        if (skipSNIApplication.get()) {
+            LOG.log(Level.FINE, "Skipping application of SNI because IBM JDK is missing setHost() method.");
+            return socket;
+        }
+
         if (socket instanceof SSLSocket) {
             try {
                 Method setHostMethod = AccessController.doPrivileged(new PrivilegedExceptionAction<Method>() {
+                    @Override
                     public Method run() throws NoSuchMethodException {
                         return socket.getClass().getMethod("setHost", String.class);
                     }
                 });
 
                 setHostMethod.invoke(socket, hostname);
-                log.finest("Applied SNI to socket for: " + hostname);
-            } catch (Exception e) {
-                log.log(Level.WARNING, "Failed to apply SNI to SSLSocket", e);
+                LOG.log(Level.FINEST, "Applied SNI to socket for host {0}", hostname);
+            } catch (PrivilegedActionException e) {
+                if (e.getCause() instanceof NoSuchMethodException && Environment.IS_IBM_JAVA) {
```

Comment:
```suggestion
                if (e.getCause() instanceof NoSuchMethodException) {
```

Reply:
@hmlnarik I am not sure we want this. I agree we should set `skipSNIApplication` to `true` for all JDKs in case the exception is `NoSuchMethod`, but we don't want to set log level to FINE for each JDK as for now we know only for IBM JDK that it is ok to skip SNI application. Does it make sense?

Reply:
Ok, let's leave this as is.